standard/classes.md

```diff
@@ -2920,7 +2943,11 @@ class Customer
 
 ### 15.6.10 Extension methods
 
-When the first parameter of a method includes the `this` modifier, that method is said to be an ***extension method***. Extension methods shall only be declared in non-generic, non-nested static classes. The first parameter of an extension method may have no modifiers other than `this`.
+When the first parameter of a method includes the `this` modifier, that method is said to be an ***extension method***. Extension methods shall only be declared in non-generic, non-nested static classes. The first parameter of an extension method is restricted, as follows:
+
+- It may have the parameter modifier `in` provided the parameter has a value type
+- It may have the parameter modifier `ref` provided the parameter has a value type or is a generic type constrained to struct
+- It may not be a pointer type.
```

Comment:
Shall rather than may? ("May not" *always* feels ambiguous to me, between "must not" and "need not". This case is "must not" but if I were to say "I may not attend the next meeting" that would be very different...) Arguably it's worth a trawl through the whole standard for "may not" at some point.

Reply:
Fixed here, but not throughout the standard